ringhold/src/main/java/org/opencadc/ringhold/InventoryValidator.java

```diff
@@ -80,25 +86,74 @@
  */
 public class InventoryValidator implements Runnable {
     private static final Logger log = Logger.getLogger(InventoryValidator.class);
-    
-    private static final String FILTER_FILENAME = "artifact-deselector.sql";
 
-    private final ArtifactDAO iteratorDAO;
+    private final ArtifactDAO artifactIteratorDAO;
     private final ArtifactDAO artifactDAO;
+    private final String deselector;
     
     public InventoryValidator(Map<String, Object> txnConfig, Map<String, Object> iterConfig) { 
         this.artifactDAO = new ArtifactDAO();
         artifactDAO.setConfig(txnConfig);
         
-        this.iteratorDAO = new ArtifactDAO();
-        iteratorDAO.setConfig(iterConfig);
-        
-        // TODO: read FILTER_FILENAME, check and fail here
-        
+        this.artifactIteratorDAO = new ArtifactDAO();
+        artifactIteratorDAO.setConfig(iterConfig);
+
+        ArtifactDeselector artifactDeselector = new ArtifactDeselector();
+        try {
+            this.deselector = artifactDeselector.getConstraint();
+        } catch (ResourceNotFoundException ex) {
+            throw new IllegalArgumentException("missing required configuration: "
+                                                   + ArtifactDeselector.SQL_FILTER_FILE_NAME, ex);
+        } catch (IOException ex) {
+            throw new IllegalArgumentException("unable to read config: " + ArtifactDeselector.SQL_FILTER_FILE_NAME, ex);
+        }
     }
 
+    /**
+     * Find an artifact with a uri pattern in the deselector,
+     * delete the artifact and generate a deleted storage location event.
+     */
     @Override
     public void run() {
-        throw new UnsupportedOperationException("NOT IMPLEMENTED");
+        final TransactionManager transactionManager = this.artifactDAO.getTransactionManager();
+        final DeletedStorageLocationEventDAO deletedStorageLocationEventDAO =
+            new DeletedStorageLocationEventDAO(this.artifactDAO);
+
+        try (final ResourceIterator<Artifact> artifactIterator =
+            this.artifactIteratorDAO.iterator(this.deselector, null)) {
+            while (artifactIterator.hasNext()) {
+                Artifact deselectorArtifact = artifactIterator.next();
+                log.debug("START: Process Artifact " + deselectorArtifact.getID() + " " + deselectorArtifact.getURI());
+
+                try {
+                    transactionManager.startTransaction();
```

Comment:
The bulk cleanup logic task (CADC-8663) specifies three things that have to happen in this transaction. Only two of them are here.

In addition, ringhold relies on an external process to clean up files. While we envisioned AD cleanup to be manual, in other sites that use ringhold to remove a set of artifacts it would be tantar. While looking at issues, #216 reminded me that there is a way to explicitly specify that a stored file is obsolete. So: rionghold will be more generally useful if it also creates an ObsoleteStorageLocation and persists it. This is a mechanism outside the model but supported in the db lib (see minoc DeleteAction) to be added to this txn (create and store an OSL).